easybuild/framework/easyblock.py

```diff
@@ -3466,8 +3472,15 @@ def make_checksum_lines(checksums, indent_level):
 
                     for key, val in sorted(ext_options.items()):
                         if key != 'checksums' and val != exts_default_options.get(key):
-                            val = quote_str(val, prefer_single_quotes=True)
-                            exts_list_lines.append("%s'%s': %s," % (INDENT_4SPACES * 2, key, val))
+                            strval = quote_str(val, prefer_single_quotes=True)
+                            line = "%s'%s': %s," % (INDENT_4SPACES * 2, key, strval)
```

Comment:
When are we dropping python 2 support? This would look so much better with f-strings :-P

Reply:
It's not *that* bad... It's not pretty, sure, but does it have to be?

We use `%s` string templating pretty consistently across the codebase, which is more important imho...

When we drop Python 2 support (perhaps with EasyBuild 5, time will tell), I'll count on you to port all of that to f-strings so we can keep things consistent. ;)

Would this be better?

```python
line = INDENT_4SPACES * 2 + "'" + key + "': " + strval + ","

Reply:
That's much more readable IMO, but I am ok keeping the other approach. My comment was a general comment, nothing that have to be necessarily changed now.

Reply:
Nah, the "%s'%s': %s," is much easier to read. I'm a C-programmer by heart....